src/main/java/edu/cornell/kfs/concur/service/impl/ConcurReportsServiceImpl.java

```diff
@@ -197,9 +204,32 @@ private String buildWorkflowUpdateXML(String workflowURI, String action, String
         String xml = "<WorkflowAction xmlns=\"" + getNamespace(workflowURI)
                 + "\"><Action>" + action + "</Action><Comment>" + comment
                 + "</Comment></WorkflowAction>";
+        
+        LOG.info("buildWorkflowUpdateXML(): Update workflow xml: " + xml);
         return xml;
     }
     
+
+    private String addConcurMessageHeaderAndTruncate(String message, int maxLength) {
+        String errorMessagesString = addMessageHeader(message);
+        errorMessagesString = truncateMessageLength(errorMessagesString, maxLength);
+        return errorMessagesString;
+    }
+
+    private String addMessageHeader(String message) {
+        if (message.length() > 0) {
+            message = ConcurConstants.ERROR_MESSAGE_HEADER + message;
+        }
+        return message;
+    }
+
+    private String truncateMessageLength(String message, int maxLength) {
+        if (message.length() > maxLength) {
+            message = message.substring(0, maxLength + 1);
```

Comment:
This line will actually return a substring of length maxLength + 1. If the substring indeed needs to be no longer than maxLength, then replace "maxLength + 1" with just "maxLength".